File: Cpp/1365_How_Many_Numbers_Are_Smaller_Than_the_Current_Number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> result;
        for(int i = 0; i < nums.size(); i++){
            int temp = 0;
            for(int j = 0; j < nums.size(); j++){
                if(nums[i] > nums[j]){
                    temp++;
                }
            }
            result.push_back(temp);
        }
        return result;
    }
};

File: Python3/1365_How_Many_Numbers_Are_Smaller_Than_the_Current_Number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        copy = sorted(nums)
        dic = {}
        for i in range(0, len(copy)):
            if (copy[i] not in dic):
                dic[copy[i]] = i
        
        result = []
        for num in nums:
            result.append(dic[num])
        
        return result
        
        """
        result = []
        for num1 in nums:
            temp = 0
            for num2 in nums:
                if (num1 > num2):
                    temp += 1
            result.append(temp)
        return result
        """
